src/net/dns.rs

```diff
@@ -0,0 +1,551 @@
+/*
+ copyright: (c) 2013-2020 by Blockstack PBC, a public benefit corporation.
+
+ This file is part of Blockstack.
+
+ Blockstack is free software. You may redistribute or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License or
+ (at your option) any later version.
+
+ Blockstack is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY, including without the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with Blockstack. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+use std::collections::VecDeque;
+
+use std::sync::mpsc::sync_channel;
+use std::sync::mpsc::SyncSender;
+use std::sync::mpsc::Receiver;
+use std::sync::mpsc::TrySendError;
+use std::sync::mpsc::TryRecvError;
+use std::sync::mpsc::RecvError;
+use std::sync::mpsc::RecvTimeoutError;
+
+use std::hash::{Hash, Hasher};
+use std::net::ToSocketAddrs;
+
+use net::PeerAddress;
+use net::Neighbor;
+use net::NeighborKey;
+use net::Error as net_error;
+use net::db::PeerDB;
+use net::asn::ASEntry4;
+
+use net::*;
+use net::codec::*;
+
+use net::StacksMessage;
+use net::StacksP2P;
+use net::GetBlocksInv;
+use net::BLOCKS_INV_DATA_MAX_BITLEN;
+use net::connection::ConnectionP2P;
+use net::connection::ReplyHandleP2P;
+use net::connection::ConnectionOptions;
+
+use net::neighbors::MAX_NEIGHBOR_BLOCK_DELAY;
+
+use net::db::*;
+
+use net::p2p::PeerNetwork;
+
+use util::sleep_ms;
+use util::db::Error as db_error;
+use util::db::DBConn;
+use util::secp256k1::Secp256k1PublicKey;
+use util::secp256k1::Secp256k1PrivateKey;
+
+use chainstate::burn::BlockHeaderHash;
+use chainstate::burn::db::burndb;
+use chainstate::burn::db::burndb::BurnDB;
+use chainstate::burn::db::burndb::BurnDBTx;
+use chainstate::burn::BlockSnapshot;
```

Comment:
It looks like there are a lot of unused imports in this file, should we clean this?